core/src/main/java/org/apache/iceberg/MetadataTableUtils.java

```diff
@@ -30,11 +30,25 @@ public static boolean hasMetadataTableName(TableIdentifier identifier) {
     return MetadataTableType.from(identifier.name()) != null;
   }
 
+  public static Table createMetadataTableInstance(Table table, MetadataTableType type) {
+    if (table instanceof BaseTable) {
+      TableOperations ops = ((BaseTable) table).operations();
+      return createMetadataTableInstance(ops, table, metadataTableName(table.name(), type), type);
+    } else {
+      throw new IllegalArgumentException(String.format("Cannot create metadata table for table: %s", table));
```

Comment:
Could we clarify this? I know we had a pr around this error message before when table instanceOf[MetadataTable] or something like that.  A user had plugged in "table.files" instead of "table" into an action and gotten a message like this. I'm actually not sure what would be better in this case, "Cannot create load metadata for non-base table %s?"

Reply:
Will do.